### DISCUSSION TITLE
How to get additional profile types such as allocated memory and heap profiling in Parca UI

### DISCUSSION THREAD

**Discussion by Adib234** [parca-dev/parca#4679] [Q&A]
Hello, I came across the tutorial and read the note regarding how Parca Agent's automatic profiling with zero-instrumentation currently only supports CPU profiling. Does this mean I would need a pull-based approach of collecting profiles in  pprof format from HTTP endpoints and how would this be done? I checked out the example https://github.com/polarsignals/pprof-example-app-go and then went on Parca UI but I still see only CPU samples being the only available profile type. I've had the example deployed in a cluster and deployed a Prometheus-based collector.

**Comment by brancz** [comment]
> I've had the example deployed in a cluster and deployed a Prometheus-based collector.

Can you share what you tried here? This is the correct approach, the scraper is the Parca server itself.

**Comment by Adib234** [comment]
I first tried deploying the example using `kubectl apply -f` and then setting up parca exactly how it's done in the tutorial for setting up Parca with Kubernetes https://www.parca.dev/docs/kubernetes/, then I deployed my Prometheus-based collector however for some reason I only see CPU samples as the only profile type

**Comment by brancz** [comment]
Can you share what you see on the `/targets` page?

**Reply by Adib234** [reply]
I was able to resolve this by fixing `parca.yaml`

**Reply by sojinss4u** [reply]
I am also facing the same issue. @Adib234 Can you please share what changes you made in parca.yaml to get the memory profiles?
Thanks

**Reply by brancz** [reply]
@sojinss4u can you describe what you’ve done and what you expected to see?

**Comment by Raviteja-Salendra** [comment]
I first deployed parca server and parca agent using kubernetes following the tutorial (https://www.parca.dev/docs/kubernetes),but in parca UI Iam able to see only CPU profiles but not memory profiling.When I installed parca server and parca agent using in another instance Iam able to see both CPU and memory profiling using kubernetes .Iam also facing the same problem @Adib234 has solved it seems.

**Reply by brancz** [reply]
For memory profiling you need to 1) instrument your application and 2) set up scraping as described here: https://www.parca.dev/docs/ingestion#combining-push-and-pull-based-ingestion

**Reply by Raviteja-Salendra** [reply]
Is it possible to get memory profiling directly through the Parca agent without instrumenting the application -just like how I can get CPU profiles for all running pods -so that memory profiling also works without adding instrumentation to the pods?
